Added Vial support for CherryBStudio87 RGB TKL PCB

WrinkleKrinkles opened the pull request:
Added Vial support for CherryBStudio87 RGB TKL PCB. 

Static RGB LED is still functional (saturation, hue, and brightness control), however RGB preset animations were removed to preserve EEPROM space.

List of removed features:

1. **All of the in-built RGB animations were removed** (build the firmware again with several choices of RGB animations if you want to enable them)
2. Tap dance, space cadet, combo, ~~key override, mousekey, extrakey~~, and grave escape were disabled.
3. Dynamic keycap layer count were set into 2. 

Total firmware size after build: ~~23810/28672 (83%, 4862 bytes free)~~  27264/28672 (95%, 1408 bytes free) with reenabled mousekeys and extrakeys after reviews on the disabled RGB animations and keymap layer count.

Possible issues:

1. ~~In the VIAL GUI, the keyboard is still listing about 4 layers even with the maximum layer set into 2; haven't tested the displayed layer 2, 3, and 4 on their functionality~~
2. Possible issues on F13 stop working and the left arrow key stopped working when using WKL 7U layout (based on the issues propping up on the CherryBStudio's Discord channel)

**Possible security issue:**
Built and tested the firmware using the newest .json file that fixes the issue (per 22th May, 2022). The said .json file needed to be loaded in the "Design" tab of VIA in "Load Draft Definition" 

I was not able to verify the .json file on other points. In terms of functionality, it is currently working without any issue being reported. 

To do:

- [x] Testing other available layers on Vial GUI.
Fixed; was putting the wrong code in config.h and rules.mk
- [x] Repairing the folders (the build failed because the files were meant to be moved to **cb87rgb** instead of **cb87**)
- [x] Adding additional Readme files for a "quick guide" to enable animation
Simply change the #undef to #define on the list of RGB animations
- [x] Following-up for the latest fix on .json file of the board
At the present moment, no fix had been uploaded to the support channels. 

Further testing might be required. So far in personal use and using the F13 button with other keycodes still functional. Will update if the "permanent" fix for the .json files were available (if required). 

Only tested on ANSI, F13 WK, 6.25u space layout.

WrinkleKrinkles committed:
VIAL ported to Vega

Readme.md for changes

WrinkleKrinkles committed:
Merge branch 'vial-kb:vial' into vial

WrinkleKrinkles committed:
Merge branch 'vial-kb:vial' into vial

WrinkleKrinkles committed:
Merge branch 'vial-kb:vial' into vial

WrinkleKrinkles committed:
Adding support for StudioKestra Galatea TKL

No RGB underglow version; tested using StudioKestra Galatea x GeonWorks TKL PCB.

Built and tested in 7U, WKL, F12 layout.

WrinkleKrinkles committed:
Add VIAL support for Ninjin TKL

Build file 49502 bytes.

No features were removed; firmware was tested on ANSI, F12, WK layout.

WrinkleKrinkles committed:
Merge branch 'vial-kb:vial' into vial

WrinkleKrinkles committed:
Added support for CherryBStudio87 RGB TKL.

Added Vial support for CherryBStudio87 RGB TKL PCB. 

Static RGB LED is still functional (saturation, hue, and brightness control), however RGB preset animations were removed to preserve EEPROM space.

List of removed features:
Several in-built RGB animations were removed (build the firmware again with several choices of RGB animations if you want to enable them)
Tap dance, space cadet, combo, key override, mousekey, extrakey, and grave escape were disabled.
Dynamic keycap layer count were set into 2. 

Total firmware size after build: 23810/28672 (83%, 4862 bytes free)

Possible issues:
In the VIAL GUI, the keyboard is still listing about 4 layers even with the maximum layer set into two.
From the .json file from May 22th, 2022 (latest update from CherryB Discord server), there were some issues on F13 stopped working and the left arrow key stopped working when using WKL layout; built and tested the firmware using the newest .json file that fixes the issue. Further testing might be required. 

Only tested on ANSI, F13 WK, 6.25u space layout.

xyzz commented on keyboards/cherrybstudio/cb87/keymaps/vial/rules.mk:
```diff
@@ -0,0 +1,11 @@
+EXTRAKEY_ENABLE = no
+SPACE_CADET_ENABLE = no
+GRAVE_ESC_ENABLE = no 
+#define DYNAMIC_KEYMAP_LAYER_COUNT 2
```
this is supposed to go into config.h

WrinkleKrinkles replied:
Thanks for the heads-up. Reuploading changes on the disabled RGB lights too. 

There was a typo on the #undef function, saved a little bit of memory size.

WrinkleKrinkles committed:
Updated config.h

Dynamic keymap layer count was properly inserted.

Fixed typo on fractal RGB animation disable.

WrinkleKrinkles committed:
Removed layer 2 and 3

Removed layer 2 and 3, saved about 2% of the total capacity in the build (at least on my end).

WrinkleKrinkles committed:
Tweaking features and revised

Keymap layer moved to the appropriate file. 

**Mousekey and extrakeys enabled**

WrinkleKrinkles reviewed (commented):
Revised keymap layer count and fixed typo on disabling RGB animations.

Reenabled mousekeys and extrakeys (sufficient memory size after alterations).

WrinkleKrinkles reviewed (commented):
Revised the config.h, keymap.c, and rules.mk files.

WrinkleKrinkles commented on keyboards/cherrybstudio/cb87/keymaps/vial/rules.mk:
```diff
@@ -4,8 +4,5 @@ VIAL_ENABLE = yes
-EXTRAKEY_ENABLE = no
 SPACE_CADET_ENABLE = no
 GRAVE_ESC_ENABLE = no 
-#define DYNAMIC_KEYMAP_LAYER_COUNT 2
```
Moved to config.h as requested

WrinkleKrinkles committed:
Moved to the correct folder

Uploaded the Vial changes to the WRONG folder... (supposed to be 'cb87rgb' instead of 'cb87'

Will delete all the additions related to cb87 later. Was tested using cb87rgb folder.

WrinkleKrinkles committed:
Delete vial.json

WrinkleKrinkles committed:
Delete rules.mk

WrinkleKrinkles committed:
Delete keymap.c

WrinkleKrinkles committed:
Delete config.h

WrinkleKrinkles reviewed (commented):
Moved to the appropriate folder.

Build failed because I had uploaded cb87rgb files to cb87 files.
